### PR TITLE
 Added a persistence project module. 

### PR DESCRIPTION
**Pull request by meln1k**
This PR is created on top of on https://github.com/mesosphere/usi/pull/17

JIRA issues: DCOS-47508

### CODE REVIEW

File: persistence/src/main/scala/com/mesosphere/usi/persistence/store/PersistenceStore.scala [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * An interface for a simple persistence store abstraction for the underlying Zookeeper store. Supported are basic
+  * create, read, update, delete and check methods along with transaction operations and Zookeeper specific children
+  * and sync methods.
+  */
+trait PersistenceStore {
```
Comment (zen-dog) [inline persistence/src/main/scala/com/mesosphere/usi/persistence/store/PersistenceStore.scala]: No, no. This storage interface is a detail of the `persistence-zookeeper` module and has nothing to do here.

**Review by zen-dog — CHANGES_REQUESTED** [review]
See my comment

**Comment by jeschkies** [comment]
> This PR is created on top of on #17
How about using the branch of #17 as a base to make reviewing simpler? Maybe there is another way.

File: persistence/build.gradle [hunk 2]
```diff
@@ -0,0 +1,3 @@
+dependencies {
+    compile project(':core-models')
```
Comment (zen-dog) [inline persistence/build.gradle:2]:
```suggestion
    api project(':core-models')
```

File: persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.mesosphere.usi.repository
+
+import com.mesosphere.usi.core.models.{PodId, PodRecord}
+
+import scala.concurrent.Future
+
+/**
+  * Repository that stores [[PodRecord]].
```
Comment (alenkacz) [inline persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: not sure if such scala docs makes sense but 🤷‍♀️

**Review by alenkacz — APPROVED** [review]
LGTM thanks!

File: persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  */
+trait PodRecordRepository {
+
+  /**
+    * Stores the provided [[PodRecord]] in the repository.
+    * @param record
+    * @return
+    */
+  def store(record: PodRecord): Future[Unit]
```
Comment (zen-dog) [inline persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: How about all four CRUD methods for the time being? And also, how about staying with "default names" e.g. `create`, `read` etc.?

File: persistence/build.gradle (hunk 2 above)
Comment (jeschkies) [reply persistence/build.gradle:2]: Neat, I did not know that Gradle could do this.

File: persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala (hunk 4 above)
Comment (jeschkies) [reply persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: Yeah, why not use @zen-dog's storage?

Comment (meln1k) [reply persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: @jeschkies  it is the next step, @zen-dog is operating on a lower level

File: persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+  * Repository for [[PodRecord]].
+  */
+trait PodRecordRepository {
+
+  /**
+    * Stores the provided [[PodRecord]] in the repository if it doesn't exist.
+    * @param record
+    * @return
```
Comment (jeschkies) [inline persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: What's returned when? This should not be a blocker now but we'll have to document it.

File: persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  /**
+    * Retrieves the [[PodRecord]] if it exists.
+    * @param podId
+    * @return Option(PodRecord) if it exists, None otherwise
+    */
+  def read(podId: PodId): Future[Option[PodRecord]]
+
+  /**
+    * Updates the [[PodRecord]] if it exists.
```
Comment (jeschkies) [inline persistence/src/main/scala/com/mesosphere/usi/repository/PodRecordRepository.scala]: And what happens when it does not exist?

**Review by jeschkies — APPROVED** [review]
Given you change the dependency to `api` 🙂

**Review by zen-dog — APPROVED** [review]
We also need a short README (you can use `metrics` for inspiration). But otherwise lgtm

File: persistence/build.gradle (hunk 2 above)
Comment (meln1k) [reply persistence/build.gradle:2]: It's only possible for java

**Pull request merged**